src/de/grobox/liberario/fragments/ProductDialogFragment.java

```diff
@@ -108,7 +113,20 @@ public void onClick(View v) {
 				// call listener if set
 				if(listener != null) {
 					EnumSet<Product> products = getProductsFromItems(adapter.getSelectedItems());
-					listener.onProductsChanged(products);
+                    // get the current network
+                    TransportNetwork current_network = Preferences.getTransportNetwork(getContext());
+                    String network_pref = current_network + "_selected_products";
+                    SharedPreferences sp = getContext().getSharedPreferences("pref_selected_products", 0);
```

Comment:
So i'll have to create  `	public final static String NETWORK_SELECTED_PRODUCTS = "pref_network_selected_products";`  for each network?